neard/build.rs

```diff
@@ -0,0 +1,68 @@
+use anyhow::{anyhow, Result};
+
+/// Returns value of given environment variable or error if missing.
+fn env(key: &str) -> Result<std::ffi::OsString> {
+    println!("cargo:rerun-if-env-changed={}", key);
+    std::env::var_os(key).ok_or_else(|| anyhow!("missing ‘{}’ environment variable", key))
+}
+
+/// Calls program with given arguments and returns its standard output.  If
+/// calling the program fails or it exits with non-zero exit status returns an
+/// error.
+fn command(prog: &str, args: &[&str]) -> Result<Vec<u8>> {
+    println!("cargo:rerun-if-env-changed=PATH");
+    let out = std::process::Command::new(prog)
+        .args(args)
+        .stderr(std::process::Stdio::inherit())
+        .output()?;
+    if out.status.success() {
+        Ok(out.stdout)
+    } else if let Some(code) = out.status.code() {
+        Err(anyhow!("{}: terminated with {}", prog, code))
+    } else {
+        Err(anyhow!("{}: killed by signal", prog))
+    }
+}
+
+/// Returns version read from git repository or ‘unknown’ if could not be
+/// determined.
```

Comment:
can you add the example (in documentation) of how this version looks like? (is it a hash? )